zio-http/src/main/scala/zio/http/api/Middleware.scala

```diff
@@ -260,6 +261,150 @@ object Middleware {
     }
   }
 
+  /**
+   * Adds the content type header to the response based on a ZIO effect
+   */
+  def withContentBase(contentBase: ContentBase): Middleware[Any, Unit, ContentBase] =
```

Comment:
I think you should call `.unit(<constant header>)` on all of these and provide the constant header. This indicates (in my mind) that there is no dynamic nature to the header, which is correct.

Reply:
So you mean, that the header should not be returned if it is not computed, but is given as a value when calling the middleware constructor?
Even though I kinda understand why, is there a harm in returning it? Could I use/access it at a later point to do something useful? 🤔

Reply:
@jdegoes done